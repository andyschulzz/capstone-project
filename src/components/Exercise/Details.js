import React, { useEffect, useRef, useState } from 'react'
import { useForm } from 'react-hook-form'
import ButtonBackToHome from '../common/ButtonBackToHome'
import ButtonEdit from './ButtonEdit'
import Form from './Form'
import * as S from './Form.styles'
import PropTypes from 'prop-types'
import { useParams } from 'react-router-dom'
import { BlueButton, GreenButton } from '../common/Button'

Details.propTypes = {
  handleExerciseChange: PropTypes.func.isRequired,
  exercises: PropTypes.array.isRequired,
  name: PropTypes.string,
  type: PropTypes.string,
  instructions: PropTypes.string,
  image: PropTypes.string,
}

export default function Details({ handleExerciseChange, exercises }) {
  const { id } = useParams()
  const { register, setValue, handleSubmit } = useForm()
  const [disabled, setDisabled] = useState(true)
  const [image, setImage] = useState()
  const mainRef = useRef(null)

  const selectedExercise = exercises.filter((exercise) => exercise.id === id)

  useEffect(() => {
    selectedExercise.forEach((exercise) => {
      setValue('name', exercise.name)
      setValue('type', exercise.type)
      setValue('instructions', exercise.instructions)
      setImage(exercise.image)
    })
  }, [selectedExercise, setValue])

  useEffect(() => {
    mainRef.current.focus()
  }, [disabled])

  return (
    <S.Form
      onSubmit={handleSubmit(handleChange)}
      data-testid="details"
      id="details"
    >
      <S.ButtonWrapper>
        <ButtonBackToHome />
        {(!disabled && (
          <GreenButton inactive mla>
            Save
          </GreenButton>
        )) || (
          <BlueButton mla onClick={handleChange}>
            Edit
          </BlueButton>
        )}
      </S.ButtonWrapper>
      <Form
        mainRef={mainRef}
        register={register}
        image={image}
        disabled={disabled}
      />
    </S.Form>
  )

  function handleChange(data) {
    setDisabled(!disabled)
    if (!disabled) {
      handleExerciseChange({ id, ...data })
    }
  }
}
