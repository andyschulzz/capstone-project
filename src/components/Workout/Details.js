import React from 'react'
import * as S from './Details.styles'
import PropTypes from 'prop-types'

Detail.propTypes = {
  register: PropTypes.func.isRequired,
  name: PropTypes.string,
  index: PropTypes.number.isRequired,
  sets: PropTypes.string,
  reps: PropTypes.string,
  weight: PropTypes.string,
}

export default function Detail({ register, name, index, sets, reps, weight }) {
  return (
    <>
      <S.H3>{name}</S.H3>
      <S.DetailWrapper>
        <S.Wrapper>
          <p>sets</p>
          <input
            ref={register({ required: true, min: 1 })}
            type="text"
            name={`sets[${index}]`}
            required
            pattern="[0-9]*"
            min="1"
            placeholder={sets}
          />
        </S.Wrapper>
        <span>&times;</span>
        <S.Wrapper>
          <p>reps</p>
          <input
            ref={register({ required: true, min: 1 })}
            type="text"
            name={`reps[${index}]`}
            required
            pattern="[0-9]*"
            min="1"
            placeholder={reps}
          />
        </S.Wrapper>
        <span>&times;</span>
        <S.Wrapper>
          <p>weight</p>
          <input
            ref={register()}
            type="text"
            pattern="[0-9]*"
            name={`weight[${index}]`}
            placeholder={weight}
          />
        </S.Wrapper>
      </S.DetailWrapper>
    </>
  )
}
