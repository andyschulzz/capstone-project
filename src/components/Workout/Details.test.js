import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Details from './Details.js'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)
const registerMock = jest.fn()

describe('<Details />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <Details
          register={registerMock}
          name="Arnold Press"
          index={1}
          reps="5"
          sets="5"
          weight="10"
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
