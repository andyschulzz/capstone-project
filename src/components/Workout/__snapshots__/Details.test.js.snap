// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  font-weight: 400;
}

<h3
    class="c0"
  >
    Arnold Press
  </h3>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 input {
  width: 48px;
  margin-top: 5px;
  padding-top: 3px;
  text-align: center;
  font-size: 1.2rem;
  border: 1px solid #ddd;
  border-radius: 5px;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c0 p {
  margin: 0;
  text-align: center;
  font-weight: 300;
  color: #888;
  text-transform: uppercase;
}

.c0 span {
  margin: 0 10px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        sets
      </p>
      <input
        min="1"
        name="sets[1]"
        pattern="[0-9]*"
        placeholder="5"
        required=""
        type="text"
      />
    </div>
    <span>
      ×
    </span>
    <div
      class="c1"
    >
      <p>
        reps
      </p>
      <input
        min="1"
        name="reps[1]"
        pattern="[0-9]*"
        placeholder="5"
        required=""
        type="text"
      />
    </div>
    <span>
      ×
    </span>
    <div
      class="c1"
    >
      <p>
        weight
      </p>
      <input
        name="weight[1]"
        pattern="[0-9]*"
        placeholder="10"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
